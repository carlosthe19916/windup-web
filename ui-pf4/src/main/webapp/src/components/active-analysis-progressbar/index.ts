export { ActiveAnalysisProgressbar } from "./active-analysis-progressbar";
