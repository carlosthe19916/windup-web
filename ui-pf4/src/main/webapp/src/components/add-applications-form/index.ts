export { AddApplicationsServerPathForm } from "./add-applications-serverpath-form";
export { AddApplicationsUploadFilesForm } from "./add-applications-uploadfiles-form";
export {
  AddApplicationsServerPathFormSchema,
  AddApplicationsUploadFilesFormSchema,
  AddApplicationsServerPathFormSchemaInitialValues,
  AddApplicationsUploadFilesFormInitialValues,
} from "./schema";
