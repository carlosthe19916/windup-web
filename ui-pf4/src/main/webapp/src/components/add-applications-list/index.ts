export { AddApplicationsList } from "./add-applications-list";
