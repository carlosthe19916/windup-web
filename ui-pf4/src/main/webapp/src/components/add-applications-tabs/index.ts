export {
  AddApplicationsTabs,
  AddApplicationsTabKey,
} from "./add-applications-tabs";
