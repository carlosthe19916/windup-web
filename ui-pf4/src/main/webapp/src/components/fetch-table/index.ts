export { FetchTable } from "./fetch-table";
