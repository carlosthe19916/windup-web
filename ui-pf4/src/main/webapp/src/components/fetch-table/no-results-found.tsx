import React from "react";
import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
} from "@patternfly/react-core";
import { SearchIcon } from "@patternfly/react-icons";

export interface NoResultsFoundProps {
  onClearFilters?: () => void;
}

export const NoResultsFound: React.FC<NoResultsFoundProps> = ({
  onClearFilters,
}) => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      <EmptyStateIcon icon={SearchIcon} />
      <Title headingLevel="h2" size="lg">
        No results found
      </Title>
      <EmptyStateBody>
        No results match the filter criteria. Remove all filters or clear all
        filters to show results.
      </EmptyStateBody>
      {onClearFilters && (
        <Button variant="link" onClick={onClearFilters}>
          Clear all filters
        </Button>
      )}
    </EmptyState>
  );
};
