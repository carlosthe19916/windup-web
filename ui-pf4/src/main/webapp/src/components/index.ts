export { ActiveAnalysisProgressbar } from "./active-analysis-progressbar";
export {
  AddApplicationsTabs,
  AddApplicationsTabKey,
} from "./add-applications-tabs";
export { AddRuleLabelTabs } from "./add-rulelabel-tabs";
export { AdvancedOptionsForm } from "./advanced-options-form";
export { AppPlaceholder } from "./app-placeholder";
export { ConditionalRender } from "./conditional-render";
export { CustomEmptyState } from "./custom-empty-state";
export { RuleLabelServerPathForm } from "./rulelabel-server-path-form";
export { RulesLabelsList } from "./rules-labels-list";
export { DeleteMatchModal } from "./delete-match-modal";
export {
  ExecutionStatus,
  mapStateToIcon,
  mapStateToLabel,
} from "./execution-status";
export { ExpandableCard } from "./expandable-card";
export { FetchErrorEmptyState } from "./fetch-error-empty-state";
export { SelectProjectEmptyMessage } from "./select-project-empty-message";
export { FetchTable } from "./fetch-table";
export { TableSectionOffline } from "./table-section-offline";
export { FilterToolbarItem } from "./filter-toolbar-item";
export { LoadingStep } from "./loading-step";
export { LogView } from "./log-view";
export { PackageDualList } from "./package-dual-list";
export {
  PackageSelection,
  PackageSelectionLoadingState,
} from "./package-selection";
export { PageHeader } from "./page-header";
export { ProcessImprovementIcon } from "./process-improvement-icon";
export { ProjectContextPageSection } from "./project-context-page-section";
export { ProjectContextSelector } from "./project-context-selector";
export {
  ProjectDetailsForm,
  projectDetailsFormSchema,
  projectDetailsFormInitialValue,
} from "./project-details-form";
export { SelectCard } from "./select-card";
export { ServerPathForm } from "./server-path-form";
export { SelectCardGallery } from "./select-card-gallery";
export { SimplePageSection } from "./simple-page-section";
export { SimplePagination } from "./simple-pagination";
export { SortByMenu } from "./sort-by-menu";
export { TransformationPath } from "./transformation-path";
export { UploadFilesForm } from "./upload-files-form";
export { Welcome } from "./welcome";
