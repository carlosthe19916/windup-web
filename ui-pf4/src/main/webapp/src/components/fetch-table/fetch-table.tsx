import React from "react";
import { useState } from "react";
import {
  EmptyState,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
  EmptyStateBody,
  Button,
  Bullseye,
  Spinner,
  Skeleton,
} from "@patternfly/react-core";
import {
  Table,
  TableHeader,
  TableBody,
  ICell,
  IRow,
  IActions,
  SortByDirection,
  ISortBy,
  IActionsResolver,
  IAreActionsDisabled,
} from "@patternfly/react-table";

import { FetchErrorEmptyState } from "components";

import { FetchStatus } from "store/common";
import { Constants } from "Constants";

import { NoResultsFound } from "./no-results-found";

export interface FetchTableProps {
  columns: ICell[];
  rows?: IRow[];
  actions?: IActions;
  actionResolver?: IActionsResolver;
  areActionsDisabled?: IAreActionsDisabled;
  fetchStatus: FetchStatus;
  loadingVariant?: "skeleton" | "spinner" | "none";
  fetchError?: any;
  onClearFilters?: () => void;
  onSortChange?: (sortBy: ISortBy) => void;
}

export const FetchTable: React.FC<FetchTableProps> = ({
  columns,
  rows,
  actions,
  actionResolver,
  areActionsDisabled,
  fetchStatus,
  fetchError,
  loadingVariant = "skeleton",
  onClearFilters,
  onSortChange,
}) => {
  const [tableSortBy, setTableSortBy] = useState<ISortBy>({});

  if (fetchError) {
    const rows: IRow[] = [
      {
        heightAuto: true,
        cells: [
          {
            props: { colSpan: columns.length },
            title: <FetchErrorEmptyState />,
          },
        ],
      },
    ];

    return (
      <Table aria-label="Table - error" cells={columns} rows={rows}>
        <TableHeader />
        <TableBody />
      </Table>
    );
  }

  if (fetchStatus !== "complete" && loadingVariant !== "none") {
    let rows: IRow[] = [];
    if (loadingVariant === "skeleton") {
      rows = [...Array(Constants.DEFAULT_PAGE_SIZE)].map(() => {
        return {
          cells: [...Array(columns.length)].map(() => ({
            title: <Skeleton />,
          })),
        };
      });
    } else if (loadingVariant === "spinner") {
      rows = [
        {
          heightAuto: true,
          cells: [
            {
              props: { colSpan: 8 },
              title: (
                <Bullseye>
                  <Spinner size="xl" />
                </Bullseye>
              ),
            },
          ],
        },
      ];
    }

    return (
      <Table aria-label="Table - loading" cells={columns} rows={rows}>
        <TableHeader />
        <TableBody />
      </Table>
    );
  }

  if (rows && rows.length === 0) {
    const rows: IRow[] = [
      {
        heightAuto: true,
        cells: [
          {
            props: { colSpan: columns.length },
            title: <NoResultsFound onClearFilters={onClearFilters} />,
          },
        ],
      },
    ];

    return (
      <Table aria-label="Table - empty" cells={columns} rows={rows}>
        <TableHeader />
        <TableBody />
      </Table>
    );
  }

  const onSort = (
    _: React.MouseEvent,
    index: number,
    direction: SortByDirection
  ) => {
    const newSortBy = { index, direction };

    setTableSortBy(newSortBy);
    if (onSortChange) {
      onSortChange(newSortBy);
    }
  };

  return (
    <>
      <Table
        aria-label="Table"
        cells={columns}
        rows={rows}
        actions={actions}
        actionResolver={actionResolver}
        areActionsDisabled={areActionsDisabled}
        sortBy={tableSortBy}
        onSort={onSort}
      >
        <TableHeader />
        <TableBody />
      </Table>
    </>
  );
};
