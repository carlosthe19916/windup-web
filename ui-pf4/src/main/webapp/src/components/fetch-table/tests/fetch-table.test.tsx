import React from "react";
import { mount, shallow } from "enzyme";
import { Label, Skeleton, Spinner } from "@patternfly/react-core";
import {
  ICell,
  IRow,
  IActions,
  SortColumn,
  sortable,
} from "@patternfly/react-table";
import { FetchErrorEmptyState } from "components";

import { FetchTable } from "../fetch-table";
import { NoResultsFound } from "../no-results-found";

describe("FetchTable", () => {
  const columns: ICell[] = [
    { title: "Col1", transforms: [sortable] },
    { title: "Col2" },
    { title: "Col3" },
  ];
  const rows: IRow[] = [...Array(15)].map((_, rowIndex) => {
    return {
      cells: [...Array(columns.length)].map((_, colIndex) => ({
        title: <Label>${`${rowIndex},${colIndex}`}</Label>,
      })),
    };
  });
  const actions: IActions = [
    {
      title: "Action1",
      onClick: jest.fn,
    },
    {
      title: "Action2",
      onClick: jest.fn,
    },
  ];

  it("Renders without crashing", () => {
    const wrapper = shallow(
      <FetchTable columns={columns} fetchStatus="complete" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders error", () => {
    const wrapper = mount(
      <FetchTable
        rows={rows}
        columns={columns}
        fetchStatus="complete"
        fetchError={"Any error"}
      />
    );
    expect(wrapper.find(FetchErrorEmptyState).length).toEqual(1);
  });

  it("Renders loading with skeleton", () => {
    const wrapper = mount(
      <FetchTable
        rows={rows}
        columns={columns}
        fetchStatus="inProgress"
        loadingVariant={"skeleton"}
      />
    );
    expect(wrapper.find(Skeleton).length).toBeGreaterThan(1);
  });

  it("Renders loading with spinner", () => {
    const wrapper = mount(
      <FetchTable
        rows={rows}
        columns={columns}
        fetchStatus="inProgress"
        loadingVariant={"spinner"}
      />
    );
    expect(wrapper.find(Spinner).length).toBe(1);
  });

  it("Renders empty table", () => {
    const wrapper = mount(
      <FetchTable rows={[]} columns={columns} fetchStatus="complete" />
    );
    expect(wrapper.find(NoResultsFound).length).toEqual(1);
  });

  it("Render rows with static actions", () => {
    const wrapper = mount(
      <FetchTable
        rows={rows}
        columns={columns}
        actions={actions}
        fetchStatus="complete"
      />
    );
    expect(wrapper.find(Label).length).toEqual(45); // 3 columns * 15 rows
  });

  it("Renders on sort", () => {
    const onSortMock = jest.fn();

    const wrapper = mount(
      <FetchTable
        rows={rows}
        columns={columns}
        fetchStatus="complete"
        onSortChange={onSortMock}
      />
    );

    wrapper.find(SortColumn).simulate("click");
    expect(onSortMock.mock.calls.length).toEqual(1);
  });
});
