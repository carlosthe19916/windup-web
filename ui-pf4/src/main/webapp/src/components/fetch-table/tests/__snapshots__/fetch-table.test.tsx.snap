// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchTable Renders without crashing 1`] = `
<Fragment>
  <Table
    aria-label="Table"
    borders={true}
    canSelectAll={true}
    cells={
      Array [
        Object {
          "title": "Col1",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Col2",
        },
        Object {
          "title": "Col3",
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    isStickyHeader={false}
    onSort={[Function]}
    ouiaSafe={true}
    role="grid"
    rowLabeledBy="simple-node"
    selectVariant="checkbox"
    sortBy={Object {}}
    variant={null}
  >
    <TableHeader />
    <Component />
  </Table>
</Fragment>
`;
